src/parsley/field.js

```diff
@@ -63,37 +75,53 @@ define('parsley/field', [
       return true;
     },
 
-    // Just validate field. Do not trigger any event
-    //  - `false` if there are constraints and at least one of them failed
-    //  - `true` in all other cases
+    // Just validate field. Do not trigger any event.
+    // Returns `true` iff all constraints pass, `false` if there are failures,
+    // or `null` if the result can not be determined yet (depends on a promise)
+    // Prefer using `whenValid`.
     isValid: function (force, value) {
+      return statusMapping[this.whenValid(force, value).state()];
+    },
+
+    whenValid: function (force, value) {
       // Recompute options and rebind constraints to have latest changes
       this.refreshConstraints();
       this.validationResult = true;
 
       // A field without constraint is valid
       if (!this.hasConstraints())
-        return true;
+        return $.when();
 
       // Value could be passed as argument, needed to add more power to 'parsley:field:validate'
       if ('undefined' === typeof value || null === value)
         value = this.getValue();
 
       if (!this.needsValidation(value) && true !== force)
-        return true;
-
-      // If we want to validate field against all constraints, just call Validator and let it do the job
-      var priorities = ['Any'];
-      if (false !== this.options.priorityEnabled) {
-        // Sort priorities to validate more important first
-        priorities = this._getConstraintsSortedPriorities();
-      }
-      // Iterate over priorities one by one, and validate related asserts one by one
-      for (var i = 0; i < priorities.length; i++)
-        if (true !== (this.validationResult = this.validateThroughValidator(value, this.constraints, priorities[i])))
-          return false;
-
-      return true;
+        return $.when();
+
+      var groupedConstraints = this._getGroupedConstraints();
+      var that = this;
+      var promises = [];
+
+      $.each(groupedConstraints, function(_, constraints) {
```

Comment:
I think this block which is very important lacks some comments and overall readability. (I'll take a closer look once I'll read all the other modifications, just going through files from page beginning to end)